src/OSPSuite.Infrastructure.Import/Services/SimulationPKAnalysesImporter.cs

```diff
@@ -101,34 +113,56 @@ private double convertValueToCoreValue(IDimension dimension, double valueInUnit,
          return dimension.UnitValueToBaseUnitValue(unit, valueInUnit);
       }
 
-      private QuantityPKParameter retrieveOrCreatePKParameterFor(CsvReader csv)
+      private QuantityPKParameter retrieveOrCreatePKParameterFor(CsvReader csv, IModelCoreSimulation modelCoreSimulation)
       {
          var parameterName = csv[PARAMETER_NAME];
          var quantityPath = csv[QUANTITY_PATH];
          var id = QuantityPKParameter.CreateId(quantityPath, parameterName);
-         if (!_importedPK.Contains(id))
-         {
-            var dimension = findDimensionFor(csv[UNIT]);
-            var pkParameter = new QuantityPKParameter {Name = parameterName, QuantityPath = quantityPath, Dimension = dimension};
-            _importedPK.Add(pkParameter);
-         }
+         if (_importedPK.Contains(id))
+            return _importedPK[id];
+
+
+         var dimension = findDimensionFor(parameterName, csv[UNIT]);
```

Comment:
This dimension will be retrieved based on the unit. So something like mg => MASS
But we need to save the value in Amount. 
So we create a merge DImension "AMOUNT, MASS" so that the value will be converted from mg to umol